lib/assessment.js

```diff
@@ -8,30 +8,44 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-var min = function min() {
-
-};
+var min = reduce(function(previousValue, currentValue, index, array) {
+  return previousValue < currentValue ? previousValue : currentValue;
+)};
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
-var sum = function sum() {
-
+var sum = function sum(previous) {
+  return
 };
 
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
-var any = function any(array, callback, thisArg) {
+  var every = function every(array, callback, thisArg) {
+  var result = true;
+  array.forEach(function(currentValue, index, array) {
+
+    result = result && callback(currentValue, index, array);
+  });
 
+  return result;
 };
 
+
 // Question 4
 // Write a function that returns a new array
 // made up of the array elements for which invoking
 // the callback returns a truthy value
 var select = function select(array, callback, thisArg) {
+  var result = true;
+  array.push(function(currentValue, index, array) {
+
+    result = result && callback(currentValue, index, array);
+  });
+
+  return result;
```

Comment:
The body of `every` is not a great starting point for this function.

You want to create a new variable inside the function and initialize it with an empty array. Then, as you're iterating over the array, if `callback(currentValue, index, array)` evaluates to `true`, you want to insert it into your newly created array using an array method, such as `push`. Finally, you return the variable your created that points to the array of selected elements you just built.

No credit.